File: 905.sort-array-by-parity.java
/*
 * @lc app=leetcode id=905 lang=java
 *
 * [905] Sort Array By Parity
 */
class Solution {
    public int[] sortArrayByParity(int[] a) {
        int[] b = new int[a.length];
        int even = 0;
        int odd = a.length - 1;
        for (int i = 0; i < a.length; i++) {
            if (a[i] % 2 == 0)
                b[even++] = a[i];
            else
                b[odd--] = a[i];
        }
        return b;
    }
}

File: 905.sort-array-by-parity.js
/*
 * @lc app=leetcode id=905 lang=javascript
 *
 * [905] Sort Array By Parity
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParity = function (A) {
  // mod后值小的在前面，也就是等于0的在前面
  A.sort((a, b) => {
    return (a % 2) - (b % 2);
  });

  return A;
};
// @lc code=end
